projects/js-packages/charts/src/components/line-chart/line-chart.tsx

```diff
@@ -525,57 +526,61 @@ const LineChartInternal = forwardRef< LineChartRef, LineChartProps >(
 
 					{ children }
 				</div>
-			</LineChartContext.Provider>
+			</SingleChartContext.Provider>
 		);
 	}
 );
 
+// Component type definitions for composition API
 type LineChartAnnotationComponents = {
 	AnnotationsOverlay: typeof LineChartAnnotationsOverlay;
 	Annotation: typeof LineChartAnnotation;
+	Legend: typeof Legend;
 };
 
 type LineChartBaseProps = Optional< LineChartProps, 'width' | 'height' | 'size' >;
 
 type LineChartComponent = React.ForwardRefExoticComponent<
-	LineChartBaseProps & React.RefAttributes< LineChartRef >
+	LineChartBaseProps & React.RefAttributes< SingleChartRef >
 > &
 	LineChartAnnotationComponents;
 
 type LineChartResponsiveComponent = React.ForwardRefExoticComponent<
-	LineChartBaseProps & ResponsiveConfig & React.RefAttributes< LineChartRef >
+	LineChartBaseProps & ResponsiveConfig & React.RefAttributes< SingleChartRef >
 > &
 	LineChartAnnotationComponents;
 
-const LineChart = forwardRef< LineChartRef, LineChartProps >( ( props, ref ) => {
-	const existingContext = useContext( ChartContext );
+const LineChartWithProvider = forwardRef< SingleChartRef, LineChartProps >( ( props, ref ) => {
+	const existingContext = useContext( GlobalChartsContext );
 
-	// If we're already in a ChartProvider context, don't create a new one
+	// If we're already in a GlobalChartsProvider context, render the core component directly
 	if ( existingContext ) {
 		return <LineChartInternal { ...props } ref={ ref } />;
 	}
 
-	// Otherwise, create our own ChartProvider
+	// Otherwise, wrap with our own GlobalChartsProvider
 	return (
-		<ChartProvider>
+		<GlobalChartsProvider>
 			<LineChartInternal { ...props } ref={ ref } />
-		</ChartProvider>
+		</GlobalChartsProvider>
 	);
-} ) as LineChartComponent;
+} );
 
-LineChart.displayName = 'LineChart';
-LineChart.AnnotationsOverlay = LineChartAnnotationsOverlay;
-LineChart.Annotation = LineChartAnnotation;
+LineChartWithProvider.displayName = 'LineChart';
 
-// Export unwrapped component for testing
-export { LineChart as LineChartUnresponsive };
+const LineChart = attachSubComponents( LineChartWithProvider, {
+	Legend: Legend,
```

Comment:
It would be good to expand the docs on composition, there is light documentation at http://localhost:50240/?path=/docs/js-packages-charts-types-line-chart--docs#linechartannotationsoverlay

Reply:
Good point. Will follow up on this when composition legend is supported by all chart types.